test/compress.js

```diff
@@ -162,8 +163,8 @@ function run_compress_tests() {
                 return false;
             }
             var ast = input.to_mozilla_ast();
-            var ast_as_string = U.AST_Node.from_mozilla_ast(ast).print_to_string();
-            var input_string = input.print_to_string();
+            var ast_as_string = U.AST_Node.from_mozilla_ast(ast).print_to_string(test.mozilla_ast);
+            var input_string = input.print_to_string(test.mozilla_ast);
```

Comment:
These reprint tests required the output options, but passing them `output_options` directly caused a whole lot unrelated failures.

Reply:
Probably needs a clone before passing it in?

Reply:
I tracked it down to the `readonly` boolean and comments: https://github.com/terser/terser/blob/91eb48b4a598fe094bddecec1f0d50a91c550be5/lib/output.js#L806-L807

Disabling this led me to an issue with printing shorthand properties. That's caused by a faulty from_mozilla_ast transform. We need to always return `AST_DefaultAssign` (`AssignmentPattern` come inside `ArrayPattern`, `ObjectPattern`, and a function's parameters).

After those fixes, everything is working with minimal options.